README.md

```diff
@@ -0,0 +1,47 @@
+# Sortition Pools
+
+Sortition pool is a logarithmic data structure used to store the pool of
+eligible operators weighted by their stakes. In the Keep network the stake
+consists of staked KEEP tokens. It allows to select a group of operators based
+on the provided pseudo-random seed.
+
+Each privileged application has its own sortition pool and is responsible for
+maintaining operator weights up-to-date.
+
+## In-Depth Reading
+
+To familiarize yourself with the sortition pool and it's design, we provide
+
++ [Building Intuition](docs/building-intuition.md)
++ [Implementation Details](docs/implementation-details.md)
+
+[Building Intuition](docs/building-intuition.md) starts the reader from the
+problem description and an easy-to-understand naive solution, and then works
+its way up to the current design of the sortition pool through a series of
+optimizations.
+
+[Implementation Details](docs/implementation-details.md) builds off of the
+knowledge in [Building Intuition](docs/building-intuition.md), and gets into
+the finer points about the data structure, data (de)serialization, how
+operators join/leave the pool, and how it all comes together to select a full
+group.
+
+## Important Facts
+
++ The max number of operators is `2,097,152`
+
+FIXME: fill this in with information about:
+
++ how long it takes to create a group of 100 operators
++ how much gas it takes
```

Comment:
Do we want to include the description of optimistic sortition here?

Reply:
can you elaborate on what you mean by optimistic sortition?

Reply:
Happy path:
1. Lock pool
2. Select off-chain
3. Submit result
4. Wait for potential challenges
5. Approve the result without on-chain selection (sortition)
6. Unlock the pool

Unhappy path:
1. Lock pool
2. Select off-chain
3. Submit malicious result
4. Receive a challenge, perform sortition on-chain, see that the group does not match the result
5. Reject the result, slash submitter
6. Let the result to be submitted one more time

https://github.com/keep-network/keep-core/tree/main/solidity/random-beacon#group-creation

It is used by the beacon and ecdsa to select large groups without performing the sortition on-chain in the happy path.

Reply:
ahh, I see! I'll add a section for that, and include a bullet point for that in "important facts"

Reply:
[bfd686b](https://github.com/keep-network/sortition-pools/pull/169/commits/bfd686b2fa9563cc1dff89e94e415161248dca7c)